src/meta/src/rpc/server.rs

```diff
@@ -124,10 +126,18 @@ pub async fn rpc_serve(
             .map_err(|e| anyhow::anyhow!("failed to connect etcd {}", e))?;
             let meta_store = Arc::new(EtcdMetaStore::new(client));
 
+            let pod_name = match env::var("POD_NAME") {
```

Comment:
It's not wise to assume this environment variable in our code, so why can't you just pass
  advertised_addr or host parameter?

Reply:
AFAIK in K8s, if you would like to pass pod information to the container in the pod, you should use environment variables (see [here](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/))

These are the arguments we have available for each meta pod

```yaml 
spec:
  containers:
  - args:
    - meta-node
    - --config-path
    - /risingwave/config/risingwave.toml
    - --listen-addr
    - 0.0.0.0:5690
    - --host
    - $(POD_IP)
    - --dashboard-host
    - 0.0.0.0:5691
    - --prometheus-host
    - 0.0.0.0:1250
    - --backend
    - etcd
    - --etcd-endpoints
    - etcd:2388
    command:
    - /risingwave/bin/risingwave
    env:
    - name: POD_IP
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: status.podIP
    - name: POD_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    - name: RUST_BACKTRACE
      value: full
```

We pass the same parameters to all of them and as a result the `RwConfig` is the same for each meat node. This is because we use one StatefulSet for all nodes and scale them using replicas. 

**Some alternatives:**

- We could skip the environment variable and **just use a UUID**. This should work and it is simple. The disadvantage is that we loose the information. With above approach the value of the lease has meaning and it tells us who the leader is. With a UUID it has no meaning. 
- We could use multiple different deployments/StatefulSets and pass different args to the containers. This will be messy and it will be harder to scale the meta nodes. Right now we only need to change `spec.replicas` to scale. I would really **advice against** such an approach. 
- We could use an [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) to get the pod-name via an environment variable, write that to a file, and then consume the file in the meta container. Instead of reading from env you would read a file. IMHO this is not much better
- We could use an **operator** to write an extra container param to the meta container which has the pod name. This means that all meta pods would have to wait until they are reconciled. This would slow down deployment and may be messy.